Regression: objToJSON "nonvoid function does not return a value" error is back

There was an issue in 2013, #5326, where `initObjToJSON` in `pandas/_libs/src/ujson/python/objToJSON.c` was causing a build error due to it lacking a return value.  Specifically, the issue was:

```
pandas/_libs/src/ujson/python/objToJSON.c: In function ‘initObjToJSON’:
pandas/_libs/src/ujson/python/objToJSON.c:181:1: error: control reaches end of non-void function [-Werror=return-type]
```

This was fixed at the time in #5334.  However, a recent commit, #30710, removed the return value again.  This has caused the warning/error to reappear, causing our builds of pandas 1.0.0 to fail.  The warning is appearing in your CI builds as well.  The difference is that we have it configured as an error.

Would it be possible to set a valid return value for the function?  Thank you.

--- a/doc/source/whatsnew/v1.0.1.rst
+++ b/doc/source/whatsnew/v1.0.1.rst
@@ -123,7 +123,7 @@ ExtensionArray
 
 Other
 ^^^^^
--
+- Regression fixed in objTOJSON.c fix return-type warning (:issue:`31463`)
 -
 
 .. ---------------------------------------------------------------------------

--- a/pandas/_libs/src/ujson/python/objToJSON.c
+++ b/pandas/_libs/src/ujson/python/objToJSON.c
@@ -147,7 +147,7 @@ enum PANDAS_FORMAT { SPLIT, RECORDS, INDEX, COLUMNS, VALUES };
 
 int PdBlock_iterNext(JSOBJ, JSONTypeContext *);
 
-void initObjToJSON(void) {
+void *initObjToJSON(void) {
     PyObject *mod_pandas;
     PyObject *mod_nattype;
     PyObject *mod_decimal = PyImport_ImportModule("decimal");
@@ -177,6 +177,8 @@ void initObjToJSON(void) {
 
     /* Initialise numpy API */
     import_array();
+    // GH 31463
+    return NULL;
 }
 
 static TypeContext *createTypeContext(void) {

--- a/pandas/_libs/src/ujson/python/objToJSON.c
+++ b/pandas/_libs/src/ujson/python/objToJSON.c
@@ -147,7 +147,7 @@ enum PANDAS_FORMAT { SPLIT, RECORDS, INDEX, COLUMNS, VALUES };
 
 int PdBlock_iterNext(JSOBJ, JSONTypeContext *);
 
-void *initObjToJSON(void) {
+void initObjToJSON(void) {
     PyObject *mod_pandas;
     PyObject *mod_nattype;
     PyObject *mod_decimal = PyImport_ImportModule("decimal");

